connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3840,8 +3840,8 @@ object functions {
 
   /**
    * Computes the first argument into a string from a binary using the provided character set (one
-   * of 'US-ASCII', 'ISO-8859-1', 'UTF-8', 'UTF-16BE', 'UTF-16LE', 'UTF-16'). If either argument
-   * is null, the result will also be null.
+   * of 'US-ASCII', 'ISO-8859-1', 'UTF-8', 'UTF-16BE', 'UTF-16LE', 'UTF-16', 'UTF-32', 'GB2312',
+   * 'GBK', 'GB18030', 'BIG5'). If either argument is null, the result will also be null.
```

Comment:
@yaooqinn . Sorry but I'm not sure about this. Could you shed some lights why UTF-x series is insufficient?

Reply:
Yes, UTF-x is always a good choice for users but not always sufficient for reading some materials pre-coded by other charsets

Reply:
We cannot add all regional encodings per language. Although I understand the intention, if we allow these Chinese entensions, we need to end up to support all European and Japan and Korean. I'd like to recommend to stick to the existing policies.

Reply:
Hi @dongjoon-hyun,

I understand your concern. The existing policies originate from the description of encode/decode functions. If we look back further in time, we can see that they are copied from Apache Hive. However, the documentation is inconsistent with its implementation in Hive or Spark when the legacy config is turned on. Thus, I don't think it can even be considered as our charset policies.

>  if we allow these Chinese entensions, we need to end up to support all European and Japan and Korean

I guess it's necessary to add other extensions in the future for our data sources to read characters encoded by these charsets.

Reply:
In the first version of `encode/decode` function we support a lot of charsets because we directly invoke the JDK APIs. We restrict it later so it makes sense to support all common charsets to avoid breaking changes.